oauth/authcode.go

```diff
@@ -86,15 +87,20 @@ func (ac *AuthorizationCodeTokenSource) Token() (*oauth2.Token, error) {
 	}
 
 	verifier := base64.RawURLEncoding.EncodeToString(verifierBytes)
+	var url string
 
-	// Generate a code challenge. Only the challenge is sent when requesting a
-	// code which allows us to keep it secret for now.
-	shaBytes := sha256.Sum256([]byte(verifier))
-	challenge := base64.RawURLEncoding.EncodeToString(shaBytes[:])
+	if ac.ClientSecret != "" {
+		url = fmt.Sprintf("%s?response_type=code&client_id=%s&redirect_uri=http://localhost:8484/&scope=%s", ac.AuthorizeURL, ac.ClientID, strings.Join(ac.Scopes, `%20`))
+	} else {
 
-	// Generate a URL with the challenge to have the user log in.
-	url := fmt.Sprintf("%s?response_type=code&code_challenge=%s&code_challenge_method=S256&client_id=%s&redirect_uri=http://localhost:8484/&scope=%s", ac.AuthorizeURL, challenge, ac.ClientID, strings.Join(ac.Scopes, `%20`))
+		// Generate a code challenge. Only the challenge is sent when requesting a
+		// code which allows us to keep it secret for now.
+		shaBytes := sha256.Sum256([]byte(verifier))
+		challenge := base64.RawURLEncoding.EncodeToString(shaBytes[:])
```

Comment:
It looks like when setting a client secret you are disabling the challenge code generation. Is that on purpose?

Reply:
In my case, code_challenge wasn't used by default, but maybe it will work fine for me as well, I'll try it soon. Will commit here the changes.

Reply:
Sounds good. If we need to make the code challenge optional we can do that too, I'm just not sure how an API would advertise whether to use it or not.

Reply:
No need to remove code_challenge, works just fine with it